huaweicloud/services/waf/resource_huaweicloud_waf_rule_blacklist.go

```diff
@@ -0,0 +1,169 @@
+package waf
+
+import (
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	rules "github.com/huaweicloud/golangsdk/openstack/waf_hw/v1/whiteblackip_rules"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const (
+	// PROTECTION_ACTION_BLOCK block the request
+	PROTECTION_ACTION_BLOCK = 0
+	// PROTECTION_ACTION_ALLOW allow the request
+	PROTECTION_ACTION_ALLOW = 1
+	// PROTECTION_ACTION_LOG log the request only
+	PROTECTION_ACTION_LOG = 2
+)
+
+func ResourceWafRuleBlackListV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWafRuleBlackListCreate,
+		Read:   resourceWafRuleBlackListRead,
+		Update: resourceWafRuleBlackListUpdate,
+		Delete: resourceWafRuleBlackListDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceWafRulesImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy_id": {
```

Comment:
missing **region** field

Reply:
Has been added and added to the document.